### PR TITLE
Move GradeCircle to Core

### PR DESCRIPTION
**Pull request by shichongrui**
refs: MBL-12314
affects: Student
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please run `yarn update-headers`. The following do not have the correct license header: 
* Core/CoreTests/Views/GradeCircleViewTests.swift
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-12314](https://instructure.atlassian.net/browse/MBL-12314)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 93.41% | 93.31% | 0.1%
**Student** | 92.27% | 92.27% | 0%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.65% | 92.65% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Views/GradeCircleView.swift [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            self.isHidden = true
+            return
+        }
+
+        guard submission.grade != nil else {
+            self.isHidden = true
+            return
+        }
+
+        self.isHidden = false
```
Comment (bluwave) [inline Core/Core/Views/GradeCircleView.swift]: This is a knitpick.  Using `self` in swift has an older `objective-c` feel to it.  Sometimes using `self` is required like in an `init` but most of the time it's not needed.  I think most(?) of swift our code generally follows this style of omitting self when not needed.

**Comment by bluwave** [comment]
Looks good.

File: Core/Core/Views/GradeCircleView.swift (hunk 1 above)
Comment (shichongrui) [reply Core/Core/Views/GradeCircleView.swift]: 👍 I'm not opposed to fixing this.

Comment (bluwave) [reply Core/Core/Views/GradeCircleView.swift]: Up to you if you want to fix or not.  I'm fine approving if not.

**Pull request merged**